app/src/main/java/com/github/studydistractor/sdp/distraction/FireBaseDistractionService.kt

```diff
@@ -37,8 +37,24 @@ class FireBaseDistractionService @Inject constructor(): DistractionService {
         return result
     }
 
-    override fun postDistraction(activity: Distraction) {
-        databaseRef.child(activity.name!!).setValue(activity)
+    override fun postDistraction(activity: Distraction, onSuccess: () -> Unit, onFailure: () -> Unit) {
+        databaseRef.child(activity.name!!).setValue(activity).addOnCompleteListener { task ->
+            if (task.isSuccessful) {
+                onSuccess()
+            } else {
+                onFailure()
+            }
+        }
+//        detach listeners
+        databaseRef.removeEventListener(object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                // Do nothing
+            }
+
+            override fun onCancelled(error: DatabaseError) {
+                // Do nothing
+            }
+        })

```

Comment:
```    override fun postDistraction(activity: Distraction, onSuccess: () -> Unit, onFailure: () -> Unit) {
       val listener = databaseRef.child(activity.name!!).setValue(activity).addOnCompleteListener { task ->
            if (task.isSuccessful) {
                onSuccess()
            } else {
                onFailure()
            }
        }
```

Where `listener` is the object you want to remove, by doing `databaseRef.removeEventListener(listener)` but honestly it is too much work to try to pass is back and let the viewModel destroy the listerner. Maybe we should put it as a task  for next sprint to have a better logic with these things.

Reply:
In my opinion for the moment, forget about detaching them. But create a task for next week.